### ISSUE TITLE
Annotator behaviour

### ISSUE THREAD

**Issue by hsorby** [cellml/libcellml#868]
Right now the  annotator will return specific items if the given id matches.  For example given the following: 
```
component1->setId("id1");
component1->setEncapsulationId("id2");
component2->setId("id3");
component2->setEncapsulationId("id4");
```

If I ask the annotator for a component encapsulation with id `id3` I will get back `component2` (`annotator->componentEncapsulation("id3");`.  Likewise if I were to ask for a component with id `id2` I will get back `component1` (`annotator->component("id2");`).

Is this the behaviour that we expect from the Annotator class?

**Comment by agarny** [comment]
I would think so.

**Comment by nickerso** [comment]
No, I don't expect this behaviour and we should look to fix this. Either by removing `componentEncapsulation` (and similar methods) from the annotator API or by having those methods do more than just wrap the expected return type method - i.e., check the item with that id is of the encapsulation type rather than any component.
